tools/buildsteps/windows/make-target-depends.bat

```diff
@@ -0,0 +1,22 @@
+@echo OFF
+
+pushd %~dp0\..\..\..
+set WORKSPACE=%CD%
+popd
+
+mkdir %WORKSPACE%\tools\windows\depends\build-%TARGET_PLATFORM%
+pushd %WORKSPACE%\tools\windows\depends\build-%TARGET_PLATFORM%
+
+cmake -G "%TARGET_CMAKE_GENERATOR%" -A %TARGET_CMAKE_GENERATOR_PLATFORM% -T host=x64 %TARGET_CMAKE_OPTIONS% ^
```

Comment:
If something goes wrong, it's not possible to re-run the bat files because they blindly try to generate the project and fails. This need to be a check to avoid issues.

One example, I've had to re-run the scripts so many times because the rename step fails, most likely because antivirus issues, we can't expect developers to know how to do this manually so the scripts should handle re-runs

Reply:
![image](https://user-images.githubusercontent.com/874927/68291033-46689500-0089-11ea-8ae4-c89246914632.png)

Rerunning `tools\buildsteps\windows\x64-uwp\make-depends.bat`
<details>
  <summary>I get the following</summary>
<pre>
    ==> Importing owner trust values...
    ==> Updating trust database...
    gpg: no need for a trustdb check
    SUCCESS: The process "gpg-agent.exe" with PID 32212 has been terminated.
    :: Synchronizing package databases...
    downloading mingw32.db...
    downloading mingw32.db.sig...
    downloading mingw64.db...
    downloading mingw64.db.sig...
    downloading msys.db...
    downloading msys.db.sig...
EXEC : error : duplicated database entry 'gcc-libs' [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\build\msys2\src\msys2-build\install.vcxproj] [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\msys2.vcxproj]
    resolving dependencies...
EXEC : warning : gcc-9.1.0-2 is up to date -- reinstalling [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\build\msys2\src\msys2-build\install.vcxproj] [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\msys2.v cxproj]
EXEC : warning : make-4.2.1-1 is up to date -- reinstalling [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\build\msys2\src\msys2-build\install.vcxproj] [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\msys2. vcxproj]
    looking for conflicting packages...

    Packages (3) gcc-libs-9.1.0-2  gcc-9.1.0-2  make-4.2.1-1

    Total Installed Size:  117.91 MiB
    Net Upgrade Size:        0.92 MiB

    :: Proceed with installation? [Y/n]
    checking keyring...
    checking package integrity...
    loading package files...
    checking for file conflicts...
EXEC : error : failed to commit transaction (conflicting files) [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\build\msys2\src\msys2-build\install.vcxproj] [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\ms ys2.vcxproj]
    gcc-libs: /usr/bin/msys-gfortran-5.dll exists in filesystem
    Errors occurred, no packages were upgraded.
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(148,5): error MSB3073: The command "setlocal [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\build\msys2 \src\msys2-build\install.vcxproj] [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\msys2.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(148,5): error MSB3073: "C:\Program Files\CMake\bin\cmake.exe" -DBUILD_TYPE=Release -P cmake_install.cmake [C:\code\kodi\ tools\windows\depends\build-x86_64-windows-native\build\msys2\src\msys2-build\install.vcxproj] [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\msys2.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(148,5): error MSB3073: if %errorlevel% neq 0 goto :cmEnd [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native \build\msys2\src\msys2-build\install.vcxproj] [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\msys2.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(148,5): error MSB3073: :cmEnd [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\build\msys2\src\msys2-buil d\install.vcxproj] [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\msys2.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(148,5): error MSB3073: endlocal & call :cmErrorLevel %errorlevel% & goto :cmDone [C:\code\kodi\tools\windows\depends\bui ld-x86_64-windows-native\build\msys2\src\msys2-build\install.vcxproj] [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\msys2.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(148,5): error MSB3073: :cmErrorLevel [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\build\msys2\src\msy s2-build\install.vcxproj] [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\msys2.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(148,5): error MSB3073: exit /b %1 [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\build\msys2\src\msys2- build\install.vcxproj] [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\msys2.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(148,5): error MSB3073: :cmDone [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\build\msys2\src\msys2-bui ld\install.vcxproj] [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\msys2.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(148,5): error MSB3073: if %errorlevel% neq 0 goto :VCEnd [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native \build\msys2\src\msys2-build\install.vcxproj] [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\msys2.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(148,5): error MSB3073: :VCEnd" exited with code -1. [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\buil d\msys2\src\msys2-build\install.vcxproj] [C:\code\kodi\tools\windows\depends\build-x86_64-windows-native\msys2.vcxproj]
</pre>
</details>

Reply:
The first error is strange, as the configure command always specifies the source directory. Have you changed that command?

The second is caused by the msys2 project code, not by re-running the configure command. I'll have a look at it.

Reply:
The msys2 error is now fixed.

Reply:
msys2 error is fixed, the other one went away after cleaning repo and starting over, likely something broken after all my testing
💯